tests/builder/Data/ByteString/Builder/Tests.hs

```diff
@@ -637,333 +641,353 @@ testsASCII =
+          [ ( coerceWord32ToFloat 0x5D1502F9 , "6.7108864e17" )
+          , ( coerceWord32ToFloat 0x5D9502F9 , "1.3421773e18" )
+          , ( coerceWord32ToFloat 0x5e1502F9 , "2.6843546e18" )
+          ]
+    , testMatches "f2sOutputLength" floatDec show
+          [ ( 1.0            , "1.0" )
+          , ( 1.2            , "1.2" )
+          , ( 1.23           , "1.23" )
+          , ( 1.234          , "1.234" )
+          , ( 1.2345         , "1.2345" )
+          , ( 1.23456        , "1.23456" )
+          , ( 1.234567       , "1.234567" )
+          , ( 1.2345678      , "1.2345678" )
+          , ( 1.23456735e-36 , "1.23456735e-36" )
+          ]
+    , testMatches "f2sPowersOf10" floatDec show $
+          fmap asShowRef [read ("1.0e" ++ show x) :: Float | x <- [-46..39 :: Int]]
+    ]
+  , testGroup "Double"
+    [ testMatches "d2sBasic" doubleDec show
+          [ ( 1.0    , "1.0" )
+          , ( (-1.0) , "-1.0" )
+          ]
+    , testMatches "f2sNonNumbersAndZero" doubleDec show
+          [ ( 0.0    , "0.0" )
+          , ( (-0.0) , "-0.0" )
+          , ( (0/0)  , "NaN" )
+          , ( (1/0)  , "Infinity" )
+          , ( (-1/0) , "-Infinity" )
+          ]
+    , testMatches "d2sSubnormal" doubleDec show
+          [ ( 2.2250738585072014e-308 , "2.2250738585072014e-308" )
+          ]
+    , testMatches "d2sMinAndMax" doubleDec show
+          [ ( (coerceWord64ToDouble 0x7fefffffffffffff) , "1.7976931348623157e308" )
+          , ( (coerceWord64ToDouble 0x0000000000000001) , "5.0e-324" )
+          ]
+    , testMatches "d2sTrailingZeros" doubleDec show
+          [ ( 2.98023223876953125e-8 , "2.9802322387695313e-8" )
+          ]
+    , testMatches "d2sRegression" doubleDec show
+          [ ( (-2.109808898695963e16) , "-2.1098088986959632e16" )
+          , ( 4.940656e-318           , "4.940656e-318" )
+          , ( 1.18575755e-316         , "1.18575755e-316" )
+          , ( 2.989102097996e-312     , "2.989102097996e-312" )
+          , ( 9.0608011534336e15      , "9.0608011534336e15" )
+          , ( 4.708356024711512e18    , "4.708356024711512e18" )
+          , ( 9.409340012568248e18    , "9.409340012568248e18" )
+          , ( 1.2345678               , "1.2345678" )
+          , ( 1.9430376160308388e16   , "1.9430376160308388e16" )
+          , ( (-6.9741824662760956e19), "-6.9741824662760956e19" )
+          , ( 4.3816050601147837e18   , "4.3816050601147837e18" )
+          ]
+    , testExpected "d2sScientific" (formatDouble scientific)
+          [ ( 0.0         , "0.0e0"         )
+          , ( 1.2345678   , "1.2345678e0"   )
+          , ( 4.294967294 , "4.294967294e0" )
+          , ( 4.294967295 , "4.294967295e0" )
+          ]
+    , testGroup "d2sStandard"
+      [ testCase "specific" do
+          singleMatches (formatDouble (standard 2)) (flip (showFFloat (Just 2)) []) ( 12.3    , "12.30"    )
+          singleMatches (formatDouble (standard 2)) (flip (showFFloat (Just 2)) []) ( 12.345  , "12.34"    )
```

Comment:
I know you've just re-formatted this test rather than introduced it, but it looks to me like this enshrining a bug: The closest `Double` to `12.345` represents a real number that is strictly greater than `12.345`, so this should round up to produce an output of `"12.35"`. (`toRational @Double 12.345 > 12.345` is `True`.)

Reply:
I added an issue to discuss this #679 because this seems unrelated to what I'm trying to accomplish here because it is a breaking change.